tests/performance/test_performance.cpp

```diff
@@ -393,16 +410,36 @@ class Test {
     }
 };
 
+bool
+valid_and_extrcat_top_k(const std::string& input, int64_t& number) {
+    std::regex pattern(R"(^search:(\d+)$)");
+    std::smatch match;
+    if (std::regex_match(input, match, pattern)) {
+        number = std::stoi(match[1].str());
+        if (number <= 0 || number > 100) {
+            std::cerr << "top k must be set to a value between 1 and 100" << std::endl;
+            exit(-1);
+        }
+        return true;
+    }
+    if (input == "search") {
+        number = 1;
+        return true;
+    }
+    return false;
+}
+
 nlohmann::json
 run_test(const std::string& dataset_path,
          const std::string& process,
          const std::string& index_name,
          const std::string& build_parameters,
          const std::string& search_parameters) {
+    int64_t top_k;
     if (process == "build") {
         return Test::Build(dataset_path, index_name, build_parameters);
-    } else if (process == "search") {
-        return Test::Search(dataset_path, index_name, build_parameters, search_parameters);
+    } else if (valid_and_extrcat_top_k(process, top_k)) {
```

Comment:
Why not check `process == "search" `?

Reply:
We check the condition of `process == "search"` in `valid_and_extrcat_top_k(process, top_k)`.

Reply:
We check the condition of `process == "search"` in `valid_and_extrcat_top_k(process, top_k)`.